### PR TITLE
Adding tesmpest test case for management cleaning steps.

### PR DESCRIPTION
**Pull request by ameya-r**
Test cases includes cleaning steps :
     - reset_idrac
     - clear_job_queue
     - known_good_state

### CODE REVIEW

File: ironic_tempest_plugin/tests/scenario/baremetal_standalone_manager.py [hunk 1]
```diff
@@ -380,6 +380,22 @@ def check_manual_partition_cleaning(self, node):
         # was actually removing the metadata, because there was not a good
         # way to achieve that check for vms and baremetal
 
+    def check_management_cleaning_wholedisk(self, node, step):
+        """Tests the cleanup step for management cleaning steps.
```
Comment (mpardhi23) [inline ironic_tempest_plugin/tests/scenario/baremetal_standalone_manager.py:384]: we can remove "steps" from docstring.

File: ironic_tempest_plugin/tests/scenario/baremetal_standalone_manager.py [hunk 2]
```diff
@@ -380,6 +380,22 @@ def check_manual_partition_cleaning(self, node):
...
+
+        :param node: Ironic node to associate instance_uuid with, it is
```
Comment (mpardhi23) [inline ironic_tempest_plugin/tests/scenario/baremetal_standalone_manager.py:386]: need to verify the node state before test case execution and after execution

File: ironic_tempest_plugin/tests/scenario/baremetal_standalone_manager.py [hunk 3]
```diff
@@ -380,6 +380,22 @@ def check_manual_partition_cleaning(self, node):
         # was actually removing the metadata, because there was not a good
         # way to achieve that check for vms and baremetal
 
+    def check_management_cleaning_wholedisk(self, node, step):
```
Comment (mpardhi23) [inline ironic_tempest_plugin/tests/scenario/baremetal_standalone_manager.py:383]: we can modify function name

File: ironic_tempest_plugin/tests/scenario/ironic_standalone/test_cleaning.py [hunk 4]
```diff
@@ -166,3 +166,28 @@ def test_software_raid(self):
         # and remove it before exiting the test.
         self.remove_root_device_hint()
         self.terminate_node(self.node['uuid'], force_delete=True)
+
+
+class BaremetalCleaningIdracWholedisk(
```
Comment (mpardhi23) [inline ironic_tempest_plugin/tests/scenario/ironic_standalone/test_cleaning.py:171]: can change class name to "BaremetalManagementCleaningIdracWholedisk"

**Review by mpardhi23 — COMMENTED** [review]
few minor changes

**Comment by ameya-r** [comment]
1. **RC** :Should not the configuration of the enabled management and power interfaces be checked? Note that with the idrac driver the same protocol must be used by those two (2) interfaces. They should both be idrac-redfish or idrac-wsman
**reply** : Since these would be in separate classes, may not need to check in test case but will be checked in outside the test case.

2. **RC**: Cannot the value of mandatory_attr be overridden by this derived class?
**reply**:overrided mandatory_attr = ['driver']. Driver is checked in base class. Was able to by pass wholedisk and image_ref as mandatory_attr.

3. **RC**: I see that the already existing, upstream Ironic BIOS tempest test method, test_bios_apply_and_reset_configuration, does not have a tempest.common.utils.services decorator. I believe that test method was added more recently than most of the others in the ironic-tempest-plugin package. Since you found removing it does not adversely affect the test execution, I recommend it be deleted before submitting this upstream for review in Gerrit. You could ask a question about it in the review, citing this information.
**reply**: As discussed we will keep this and post a comment in upstream after raising the patch.

4. **RC**: after further reflecting on our discussion this past Friday, I believe separate tests/classes should be created for testing the idrac driver WS-Man (idrac-wsman) and Redfish (idrac-redfish) management interface implementations of the iDRAC cleaning step
**reply**: Created 2 seperate classes

5. **RC**: have we considered cleaning multiple baremetal nodes in parallel? Please see the test methods in class InspectorBasicTest.
**reply** : As discussed, will take this as enhancement.

6 **RC.**: Also, we could add Dell's copyright notice to the changed file
**reply**: Added in test_cleaning.py

7. **RC** : this is a friendly reminder to draft the commit message before submitting this change upstream for the community to review 
**reply** : Will be doing in actual PR or first commit of this PR

File: ironic_tempest_plugin/tests/scenario/baremetal_standalone_manager.py [hunk 5]
```diff
@@ -451,6 +467,12 @@ class BaremetalStandaloneScenarioTest(BaremetalStandaloneManager):
     # set via a different test).
     raid_interface = None
 
+    # The management interface to be use in the test
```
Comment (DhuldevValekar30) [inline ironic_tempest_plugin/tests/scenario/baremetal_standalone_manager.py:470]: use the same msg to keep it consistent with the other interfaces.

Comment (ameya-r) [reply ironic_tempest_plugin/tests/scenario/baremetal_standalone_manager.py]: Done. Added description as per other interfaces.
Also added Dell copyrights in other 2 files.

**Comment by DhuldevValekar30** [comment]
LGTM, Thanks!

**Review by DhuldevValekar30 — APPROVED** [review]

**Comment by calsoft-internal** [comment]
Since test case is merged in upstream, closing this pull request

**Pull request closed without merging**